BE/src/main/java/louie/dong/airbnb/book/BookService.java

```diff
@@ -0,0 +1,61 @@
+package louie.dong.airbnb.book;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import lombok.RequiredArgsConstructor;
+import louie.dong.airbnb.accommodation.Accommodation;
+import louie.dong.airbnb.accommodation.AccommodationRepository;
+import louie.dong.airbnb.book.dto.BookDetailResponse;
+import louie.dong.airbnb.book.dto.BookResponse;
+import louie.dong.airbnb.book.dto.BookSaveRequest;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@RequiredArgsConstructor
+@Transactional(readOnly = true)
+@Service
+public class BookService {
+
+	private final BookRepository bookRepository;
+	private final AccommodationRepository accommodationRepository;
+
+	public BookDetailResponse findById(Long id) {
+		Book book = bookRepository.findById(id).orElseThrow(NoSuchElementException::new);
+		return new BookDetailResponse(book);
+	}
+
+	public List<BookResponse> findAll() {
+		List<Book> books = bookRepository.findAll();
+		List<BookResponse> bookResponses = new ArrayList<>();
+
+		for (Book book : books) {
+			bookResponses.add(new BookResponse(book));
+		}
+		return bookResponses;
+	}
+
+	@Transactional
+	public void cancel(Long id) {
+		Book book = bookRepository.findById(id).orElseThrow(NoSuchElementException::new);
+		book.changeCanceled(true);
+	}
```

Comment:
changeCanceled를 cancel메소드로 변경하면(+메소드 분리) 아래와 같이 간단한 메소드를 만들 수 있을것 같아요.

```suggestion
	@Transactional
	public void cancel(Long id) {
		findByIdOrThrow(id).cancel();
	}
	
	public void findByIdOrThrow(long id){
		return bookRepository.findById(id)
			.orElseThrow(NoSuchElementException::new);
	}
```

Reply:
메서드명만 봐도 코드를 쉽게 이해할 수 있네요! 
왕민 말씀대로 리팩터링 해보겠습니다 😊